### PR TITLE
Link retention

### PR DESCRIPTION
**Pull request by alexduf**
The rows in the `UserSubscription` were deleted 30 months after the last interaction with a device.

This doesn't make sense, we need to delete these rows as soon as the subscription attached to it is being deleted.

This PR:

- Enable the dynamo event stream for the `Subscription` table
- creates a lambda that listen to that stream
- reacts only on TTL events from the `Subscription` table
- delete elements in the `UserSubscription` table when the corresponding row has been deleted from the `Subscription` table

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -11,10 +11,10 @@
     // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
     "sourceMap": true,                        /* Generates corresponding '.map' file. */
     // "outFile": "./",                       /* Concatenate and emit output to single file. */
-     "outDir": "tsc-target",                      /* Redirect output structure to the directory. */
+    "outDir": "tsc-target",                   /* Redirect output structure to the directory. */
     // "rootDir": "./",                       /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
     // "composite": true,                     /* Enable project compilation */
-    // "incremental": true,                   /* Enable incremental compilation */
+    "incremental": true,                      /* Enable incremental compilation */
```
Comment (NathanielBennett) [inline tsconfig.json]: What's the purpose of this change?

Comment (alexduf) [reply tsconfig.json]: Faster compilation time. Admittedly unrelated!

Comment (NathanielBennett) [reply tsconfig.json]: Fair play!
